File: sdk/archive/v1.0.81/javascript/projects.js
function stripUndef(o) { const r = {}; for (const k in o) if (o[k] !== undefined) r[k] = o[k]; return r; }

export class Projects {
  constructor(ctx) { this._ctx = ctx; }

  async createProjectAndRun(name, private_ = undefined, type = undefined, description = undefined, directory = undefined, photoUploadVector = undefined, thumbnail = undefined, runName = undefined, { outcomes, models } = {}) {
    this._ctx.requireTokens();
    let path = `/api/v1/project_outcome/`;
    const query = new URLSearchParams();
    if (outcomes !== undefined) (Array.isArray(outcomes) ? outcomes : [outcomes]).forEach(v => query.append('outcomes', v));
    if (models !== undefined) (Array.isArray(models) ? models : [models]).forEach(v => query.append('models', v));
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const body = stripUndef({
      name: name,
      private: private_,
      type: type,
      description: description,
      directory: directory,
      photo_upload_vector: photoUploadVector,
      thumbnail: thumbnail,
      run_name: runName,
    });
    const { data } = await this._ctx.client.request(path, 'POST', this._ctx.accessToken, this._ctx.refreshToken, body);
    return data;
  }

  async markProjectUploadComplete(projectTableName, { skippedFileCount } = {}) {
    this._ctx.requireTokens();
    let path = `/api/v1/project/${encodeURIComponent(projectTableName)}/upload_complete/`;
    const query = new URLSearchParams();
    if (skippedFileCount !== undefined) (Array.isArray(skippedFileCount) ? skippedFileCount : [skippedFileCount]).forEach(v => query.append('skipped_file_count', v));
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const { data } = await this._ctx.client.request(path, 'POST', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async checkProjectStatus(projectTableName) {
    this._ctx.requireTokens();
    const path = `/api/v1/project/status/${encodeURIComponent(projectTableName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getProjectPrelimModelRequestTemplate(projectTableName) {
    this._ctx.requireTokens();
    const path = `/api/v1/project_outcome/${encodeURIComponent(projectTableName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getUserProjects() {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/user`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getAdminProjects() {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/admin`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getProjectById(projectId) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/${encodeURIComponent(projectId)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getProjectByDirectory(directory) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/directory/${encodeURIComponent(directory)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async updateProject(projectId, { private: private_, type, description, directory, name, thumbnail } = {}) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/${encodeURIComponent(projectId)}`;
    const body = stripUndef({
      private: private_,
      type: type,
      description: description,
      directory: directory,
      name: name,
      thumbnail: thumbnail,
    });
    const { data } = await this._ctx.client.request(path, 'PUT', this._ctx.accessToken, this._ctx.refreshToken, body);
    return data;
  }

  async updateProjectPhotoCount(tableName, { filesFailedCount } = {}) {
    this._ctx.requireTokens();
    let path = `/api/v1/projects_photos/${encodeURIComponent(tableName)}/`;
    const query = new URLSearchParams();
    if (filesFailedCount !== undefined) (Array.isArray(filesFailedCount) ? filesFailedCount : [filesFailedCount]).forEach(v => query.append('files_failed_count', v));
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const { data } = await this._ctx.client.request(path, 'PUT', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async updateProjectCreateUploadReport(tableName, { filesFailedCount } = {}) {
    this._ctx.requireTokens();
    let path = `/api/v1/project_upload_report/${encodeURIComponent(tableName)}/`;
    const query = new URLSearchParams();
    if (filesFailedCount !== undefined) (Array.isArray(filesFailedCount) ? filesFailedCount : [filesFailedCount]).forEach(v => query.append('files_failed_count', v));
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const { data } = await this._ctx.client.request(path, 'PUT', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async requestProjectSimilarityQueue(projectTableName, level) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects_similarity_queue/${encodeURIComponent(projectTableName)}/level/${encodeURIComponent(level)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async requestProjectEvidenceQueue(projectTableName) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects_evidence_queue/${encodeURIComponent(projectTableName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async requestProjectPersonhoodQueue(projectTableName) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects_personhood_queue/${encodeURIComponent(projectTableName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async requestInsightsQueue(analysisLevel, identifier) {
    this._ctx.requireTokens();
    const path = `/api/v1/insights_queue/analysis_level/${encodeURIComponent(analysisLevel)}/identifier/${encodeURIComponent(identifier)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async requestProjectExport(projectTableName) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects_export/${encodeURIComponent(projectTableName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getProjectDataExportUploadStatus(projectTableName, modelName) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/${encodeURIComponent(projectTableName)}/upload_status/${encodeURIComponent(modelName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async addProjectEvent(projectTableName, event, detail = undefined) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/${encodeURIComponent(projectTableName)}/event`;
    const body = stripUndef({
      event: event,
      detail: detail,
    });
    const { data } = await this._ctx.client.request(path, 'POST', this._ctx.accessToken, this._ctx.refreshToken, body);
    return data;
  }

  async deleteProject(projectId) {
    this._ctx.requireTokens();
    const path = `/api/v1/projects/${encodeURIComponent(projectId)}`;
    const { data } = await this._ctx.client.request(path, 'DELETE', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }
}

File: sdk/archive/v1.5.0-dev.84/python/mediaviz_sdk/projects.py
from __future__ import annotations
from typing import Any
from urllib.parse import quote, urlencode
import httpx

from .errors import handle_response


class Projects:
    def __init__(self, ctx) -> None:
        self._ctx = ctx

    def create_project_and_run(
        self,
        name: str,
        private: bool | None = None,
        type: int | None = None,
        description: str | None = None,
        directory: str | None = None,
        photoUploadVector: int | None = None,
        thumbnail: str | None = None,
        runName: str | None = None,
        outcomes: Any | None = None,
        models: Any | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/project_outcome/'
        _q: dict[str, Any] = {}
        if outcomes is not None:
            _q['outcomes'] = outcomes
        if models is not None:
            _q['models'] = models
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        body = {k: v for k, v in {
            'name': name,
            'private': private,
            'type': type,
            'description': description,
            'directory': directory,
            'photo_upload_vector': photoUploadVector,
            'thumbnail': thumbnail,
            'run_name': runName,
        }.items() if v is not None}
        return self._ctx.client.request(path, 'POST', self._ctx.access_token, self._ctx.refresh_token, body).data

    def mark_project_upload_complete(
        self,
        project_table_name: str,
        skipped_file_count: Any | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/project/' + quote(str(project_table_name), safe='') + '/upload_complete/'
        _q: dict[str, Any] = {}
        if skipped_file_count is not None:
            _q['skipped_file_count'] = skipped_file_count
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        return self._ctx.client.request(path, 'POST', self._ctx.access_token, self._ctx.refresh_token).data

    def check_project_status(self, project_table_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/project/status/' + quote(str(project_table_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_project_prelim_model_request_template(self, project_table_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/project_outcome/' + quote(str(project_table_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_user_projects(self) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/user'
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_admin_projects(self) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/admin'
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_project_by_id(self, project_id: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/' + quote(str(project_id), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_project_by_directory(self, directory: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/directory/' + quote(str(directory), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def update_project(
        self,
        project_id: str,
        *,
        private: bool | None = None,
        type: int | None = None,
        description: str | None = None,
        directory: str | None = None,
        name: str | None = None,
        thumbnail: str | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/' + quote(str(project_id), safe='')
        body = {k: v for k, v in {
            'private': private,
            'type': type,
            'description': description,
            'directory': directory,
            'name': name,
            'thumbnail': thumbnail,
        }.items() if v is not None}
        return self._ctx.client.request(path, 'PUT', self._ctx.access_token, self._ctx.refresh_token, body).data

    def update_project_photo_count(
        self,
        table_name: str,
        files_failed_count: Any | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects_photos/' + quote(str(table_name), safe='') + '/'
        _q: dict[str, Any] = {}
        if files_failed_count is not None:
            _q['files_failed_count'] = files_failed_count
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        return self._ctx.client.request(path, 'PUT', self._ctx.access_token, self._ctx.refresh_token).data

    def update_project_create_upload_report(
        self,
        table_name: str,
        files_failed_count: Any | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/project_upload_report/' + quote(str(table_name), safe='') + '/'
        _q: dict[str, Any] = {}
        if files_failed_count is not None:
            _q['files_failed_count'] = files_failed_count
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        return self._ctx.client.request(path, 'PUT', self._ctx.access_token, self._ctx.refresh_token).data

    def request_project_similarity_queue(
        self,
        project_table_name: str,
        level: str,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects_similarity_queue/' + quote(str(project_table_name), safe='') + '/level/' + quote(str(level), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def request_project_evidence_queue(self, project_table_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects_evidence_queue/' + quote(str(project_table_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def request_project_personhood_queue(self, project_table_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects_personhood_queue/' + quote(str(project_table_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def request_insights_queue(self, analysis_level: str, identifier: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/insights_queue/analysis_level/' + quote(str(analysis_level), safe='') + '/identifier/' + quote(str(identifier), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def request_project_export(self, project_table_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects_export/' + quote(str(project_table_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_project_process_status(
        self,
        project_table_name: str,
        model_name: str,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/' + quote(str(project_table_name), safe='') + '/process_status/' + quote(str(model_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def add_project_event(
        self,
        project_table_name: str,
        event: str,
        detail: str | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/' + quote(str(project_table_name), safe='') + '/event'
        body = {k: v for k, v in {
            'event': event,
            'detail': detail,
        }.items() if v is not None}
        return self._ctx.client.request(path, 'POST', self._ctx.access_token, self._ctx.refresh_token, body).data

    def delete_project(self, project_id: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/projects/' + quote(str(project_id), safe='')
        return self._ctx.client.request(path, 'DELETE', self._ctx.access_token, self._ctx.refresh_token).data
